backend/controllers/prisonerProfile/prisonerLocationHistory.js

```diff
@@ -12,7 +12,43 @@ const {
   app: { notmEndpointUrl: dpsUrl },
 } = require('../../config')
 
-module.exports = ({ prisonApi, logError }) => async (req, res) => {
+const fetchStaffName = (context, staffId, prisonApi) =>
+  prisonApi.getStaffDetails(context, staffId).then(staff => formatName(staff.firstName, staff.lastName))
+
+const fetchReasonDescription = (context, assignmentReasonCode, caseNotesApi) => {
+  if (assignmentReasonCode) {
+    return caseNotesApi
+      .getCaseNoteTypes(context)
+      .then(caseNoteTypes => caseNoteTypes.find(type => type.code === 'MOVED_CELL'))
+      .then(cellMoveTypes => {
+        return cellMoveTypes?.subCodes.map(subType => ({
+          value: subType.code,
+          text: subType.description,
+        }))
+      })
+      .then(cellMoveReasonRadioValues =>
+        cellMoveReasonRadioValues.find(record => record.value === assignmentReasonCode)
+      )
+      .then(assignmentReason => assignmentReason.text)
+  }
+  return ''
+}
+
+const fetchWhatHappened = (
+  context,
+  offenderNo,
+  bookingId,
+  bedAssignmentHistorySequence,
+  caseNotesApi,
+  whereaboutsApi
+) =>
+  whereaboutsApi
+    .getCellMoveReason(context, bookingId, bedAssignmentHistorySequence)
+    .then(cellMoveReason => caseNotesApi.getCaseNote(context, offenderNo, cellMoveReason.cellMoveReason.caseNoteId))
+    .then(caseNote => caseNote.text)
+    .catch(err => 'Not entered')
```

Comment:
This will show false information to the user if it fails for reasons other than a 404.